src/Coordination/KeeperDispatcher.cpp

```diff
@@ -760,19 +760,36 @@ void KeeperDispatcher::clusterUpdateWithReconfigDisabledThread()
 
 void KeeperDispatcher::clusterUpdateThread()
 {
+    using enum KeeperServer::ConfigUpdateResult;
+
+    // If command was accepted, we don't need to send it to server again, just check whether we
+    // got it in our local config.
+    bool last_command_was_accepted = false;
+
     while (!shutdown_called)
     {
         ClusterUpdateAction action;
         if (!cluster_update_queue.pop(action))
             return;
 
-        if (server->applyConfigUpdate(action))
-            LOG_DEBUG(log, "Processing config update {}: accepted", action);
-        else // TODO (myrrc) sleep a random amount? sleep less?
+        switch (server->applyConfigUpdate(action, last_command_was_accepted))
         {
-            (void)cluster_update_queue.pushFront(action);
-            LOG_DEBUG(log, "Processing config update {}: declined, backoff", action);
-            std::this_thread::sleep_for(50ms);
+            case Declined: {
+                LOG_DEBUG(log, "Processing config update {}: declined, backoff", action);
+                (void)cluster_update_queue.pushFront(action);
+                break;
+            }
+            case Accepted: {
+                LOG_DEBUG(log, "Processing config update {}: accepted, backoff to check for committed", action);
+                (void)cluster_update_queue.pushFront(action);
```

Comment:
this creates infinite loop if we define with invalid host, try adding in test a case where you add nonexistent host
```
zk.reconfig(
        joining="server.4=nonexistent:2123", leaving=None, new_members=None
    )
```

Reply:
Valid point. I don't want to rely on external timeouts so I added a https://github.com/eBay/NuRaft/pull/470 small update to Nuraft in which we could track errors and wipe the config update queue